hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOMRatisSnapshots.java

```diff
@@ -1036,6 +1069,272 @@ public void testInstallCorruptedCheckpointFailure() throws Exception {
+        return getUsedBytes(newLeaderOM) ==
+            usedBytes - newKeyInfo.getDataSize();
+      } catch (IOException e) {
+        Assertions.fail();
+        return false;
+      }
+    }, 1000, 10000);
+
+    GenericTestUtils.waitFor(() -> {
+      try {
+        return Objects.isNull(omKeyInfoTable.get(newKey));
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }, 1000, 10000);
+
+    // Check whether newly created snapshot data is reclaimed
+    client.getObjectStore()
+        .deleteSnapshot(volumeName, bucketName, newSnapshot.getName());
+    GenericTestUtils.waitFor(() -> {
+      Table<String, SnapshotInfo> snapshotInfoTable =
+          newLeaderOM.getMetadataManager().getSnapshotInfoTable();
+      try {
+        return null == snapshotInfoTable.get(newSnapshot.getTableKey());
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }, 1000, 10000);
+
+    // Check whether compaction logs get appeneded to by forcning compaction
+    newLeaderOM.getMetadataManager()
+        .getStore()
+        .compactDB();
+    int newNumberOfLogFiles = 0;
+    long newContentLength;
+    try (BufferedReader bufferedReader =
+             Files.newBufferedReader(currentCompactionLog);
+         DirectoryStream<Path> files =
+             Files.newDirectoryStream(compactionLogPath)) {
+      newContentLength = bufferedReader.lines()
+          .mapToLong(String::length)
+          .reduce(0L, Long::sum);
+      for (Path ignored : files) {
+        newNumberOfLogFiles++;
+      }
+    }
+    Assertions.assertTrue(numberOfLogFiles != newNumberOfLogFiles
+        || contentLength != newContentLength);
+
+    // Check whether compaction backup files were pruned
+    final int finalNumberOfSstFiles = numberOfSstFiles;
+    GenericTestUtils.waitFor(() -> {
+      int newNumberOfSstFiles = 0;
+      try (DirectoryStream<Path> files =
+               Files.newDirectoryStream(sstBackupDirPath)) {
+        for (Path ignored : files) {
+          newNumberOfSstFiles++;
+        }
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+      return finalNumberOfSstFiles > newNumberOfSstFiles;
+    }, 1000, 10000);
+
+    // Snap diff
+    String firstSnapshot = createOzoneSnapshot(newLeaderOM,
+        snapshotNamePrefix + RandomStringUtils.randomNumeric(10)).getName();
+    String diffKey = writeKeys(1).get(0);
+    String secondSnapshot = createOzoneSnapshot(newLeaderOM,
+        snapshotNamePrefix + RandomStringUtils.randomNumeric(10)).getName();
+    SnapshotDiffReportOzone diff = getSnapDiffReport(volumeName, bucketName,
+        firstSnapshot, secondSnapshot);
+    Assertions.assertEquals(Collections.singletonList(
+            SnapshotDiffReportOzone.getDiffReportEntry(
+                SnapshotDiffReport.DiffType.CREATE, diffKey, null)),
+        diff.getDiffList());
+  }
+
+  private long getUsedBytes(OzoneManager newLeaderOM) throws IOException {
```

Comment:
nit: just call it `leaderOm` or `ozoneManager`.
```suggestion
  private long getUsedBytes(OzoneManager leaderOm) throws IOException {
```